stdlib/collections/__init__.pyi

```diff
@@ -266,15 +266,14 @@ class Counter(dict[_T, int], Generic[_T]):
     def subtract(self, __mapping: Mapping[_T, int]) -> None: ...
     @overload
     def subtract(self, __iterable: Iterable[_T]) -> None: ...
-    # The Iterable[Tuple[...]] argument type is not actually desirable
-    # (the tuples will be added as keys, breaking type safety) but
-    # it's included so that the signature is compatible with
-    # Dict.update. Not sure if we should use '# type: ignore' instead
-    # and omit the type from the union.
-    @overload
+    # The second overload is deliberately different to dict.update()
+    # (if it were `Iterable[_T] | Iterable[tuple[_T, int]]`,
+    # the tuples would be added as keys, breaking type safety),
+    # hence the type: ignore[override]
+    @overload  # type: ignore[override]
     def update(self, __m: SupportsKeysAndGetItem[_T, int], **kwargs: int) -> None: ...
```

Comment:
`SupportsKeysAndGetItem` doesn't seem correct. The implementation does `if isinstance(iterable, _collections_abc.Mapping)` and then calls `.items()`.

I initially added `SupportsKeysAndGetItem` to `dict.__init__` to get a tkinter thing to work, and apparently it has now found its way here somehow :)

Reply:
Oops, I think this one was me. I'll change it back to `Mapping`.